imports/ui/AddTask.js

```diff
@@ -35,61 +36,82 @@ export class AddTask extends React.Component {
     this.setState({ dueDate });
   };
 
+  // eslint-disable-next-line no-undef
+  handleSubmitTaks = () => !!this.state.description;
+
   // eslint-disable-next-line no-undef
   addTaskAndGo = () => {
-    const { router: { history } } = this.context;
-    const { addTask } = this.props;
-    addTask({
-      variables: {
-        task: {
-          ...this.state,
+    if (this.handleSubmitTaks()) {
+      const { addTask } = this.props;
+      addTask({
+        variables: {
+          task: {
+            _id: this.state._id,
+            description: this.state.description,
+            details: this.state.details,
+            dueDate: this.state.dueDate,
+            done: this.state.done,
+          },
         },
-      },
-    }).then(({ data: { addTask: { _id } } }) => {
-      if (_id) {
-        history.push('/');
-      }
+      })
+        .then(() => {})
+        .catch(error => {
+          // eslint-disable-next-line no-console
+          console.log(error);
+        });
+      document.getElementById('bottomNavigationActionTasks').click();
+    } else {
+      this.setState({ error: true });
+    }
+  };
+
+  // eslint-disable-next-line no-undef
+  callBack = () => {
+    this.setState({
+      error: false,
     });
   };
 
   render() {
     return (
-      <form className="form">
-        <TextField
-          name="description"
-          label="Description"
-          value={this.state.description}
-          onChange={this.onInputChange}
-          fullWidth
-        />
-        <TextField
-          name="details"
-          label="Details"
-          value={this.state.details}
-          onChange={this.onInputChange}
-          fullWidth
+      <Fragment>
+        <form className="form">
+          <TextField
+            name="description"
+            label="Description"
+            value={this.state.description}
+            onChange={this.onInputChange}
+            fullWidth
+            required
+          />
+          <TextField
+            name="details"
+            label="Details"
+            value={this.state.details}
+            onChange={this.onInputChange}
+            fullWidth
+          />
+          <DatePicker
+            value={this.state.dueDate}
+            returnMoment={false}
+            onChange={this.handleDueDateChange}
+            minDate={new Date()}
+          />
+          <Button
+            className="form-action"
+            raised
+            color="primary"
+            onClick={this.addTaskAndGo}
+          >
+            <Save />
+          </Button>
+        </form>
+        <Error
```

Comment:
It should be specific for each error because any error will show the same message